src/mine-sweeper.tsx

```diff
@@ -176,16 +180,40 @@ export default class MineSweeper extends React.Component<IMineSweeperProps, IMin
     }
   }
 
-  private revealCell(x: number, y: number) {
-    const mineField = this.state.mineField;
-    mineField[x][y].isRevealed = true;
-    mineField[x][y].isFlagged = false;
-    this.setState({
-      mineField
-    });
-    this.checkSurroundingCells(x, y);
+  private revealCell = (x: number, y: number) => {
+    const field = this.state.mineField[x][y]
+    this.revealCellByMineField(field)
+    if (field.mineCounter === 0) {
+      this.revealSurroundingCells(x, y)
+    }
   }
 
+  private revealCellByMineField = (cell: IMineField) =>
+  {
+    if (cell.isMine) { return }
```

Comment:
this is an issue, we wouldn't be able to reveal mines correctly like this

Reply:
revealCellByMineField is called during auto reveal operation,
the function includes that conditional so we don't auto reveal the cell that has mine in it.